gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp

```diff
@@ -0,0 +1,444 @@
+    && nGeometryElementDepth == 0 )
+    {
+        const char *pszTag = XMLTagSplit(pszName);
+        
+        StopDataCollect();
+        if ( !osElementString.empty() ) 
+        {
+            if( bSchemaOlny )
+            {
+                OGRFieldType eType = OFTString;
+                OGRFieldDefn oField(pszTag, eType);
+                poFeatureDefn->AddFieldDefn(&oField);
+            }
+            else
+            {
+                const int iFieldIndex = poFeatureDefn->GetFieldIndex(pszTag);
+                if( iFieldIndex > -1 )
+                {
+                    poFeature->SetField(iFieldIndex, osElementString.c_str());
+                }
+            }
+        }
+        osElementString.Clear();
+    }
+    else if( nAttributeElementDepth == nCurrentDepth )
+    {
+        nAttributeElementDepth = 0;
+    }
+    else if( nGeometryElementDepth > 0 && nCurrentDepth > nGeometryElementDepth )
+    {
+        osElementString += "</";
+        osElementString += pszName;
+        osElementString += ">";
+    }
+    else if( nGeometryElementDepth == nCurrentDepth )
+    {
+        StopDataCollect();
+        if( !osElementString.empty() )
+        {
+            std::unique_ptr<OGRGeometry> poGeom = std::unique_ptr<OGRGeometry>{
+                CPL_REINTERPRET_CAST(OGRGeometry *, OGR_G_CreateFromGML(osElementString.c_str())) };
+            if( poGeom && !poGeom->IsEmpty() )
+            {
+                if( !bSchemaOlny )
+                    poFeature->SetGeometryDirectly(poGeom.release());
+                else
+                {
+                    OGRGeomFieldDefn *poGeomField = poFeatureDefn->GetGeomFieldDefn(0);
+                    poGeomField->SetSpatialRef(poGeom->getSpatialReference());
+                    poGeomField->SetType(poGeom->getGeometryType());
+                }
+            }
+            else
+            {
+                CPLError(CE_Failure, CPLE_AppDefined, "Parsing geometry as GML failed");
+                XML_StopParser(oParser.get(), XML_FALSE);
+            }
+        }
+
+        osElementString.Clear();
+        nGeometryElementDepth = 0;
+    }
+    else if( nFeatureElementDepth == nCurrentDepth )
+    {
+        nFeatureElementDepth = 0;
+
+        if( !bSchemaOlny )
+            poFeature->SetFID(nNextFID++);
+
+        XML_StopParser(oParser.get(), XML_TRUE);
+    }
+    else if( nFeatureCollectionDepth == nCurrentDepth )
+    {
+        nFeatureCollectionDepth = 0;
+    }
+}
+
+/************************************************************************/
+/*                          ConfigureParser()                           */
+/************************************************************************/
+
+void OGRLVBAGLayer::ConfigureParser()
+{
+    const auto startElementWrapper = [](void *pUserData, const char *pszName, const char **ppszAttr)
+    {
+        CPL_STATIC_CAST(OGRLVBAGLayer*, pUserData)->StartElementCbk(pszName, ppszAttr);
```

Comment:
plain static_cast<> would be preferable (here and in all other places). CPL_STATIC_CAST is for code that can be build as .c or .cpp .